### PR TITLE
QuantumGuard application submission

### PR DESCRIPTION
**Pull request by andredif**
#### Project Abstract

QuantumGuard, a quantum-safe blockchain based on Substrate framework and NIST PQC.
The grant aims to provide a quantum-safe node and a custom browser wallet extension.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ Polkadot (USDC & USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by andredif** [comment]
I have read and hereby sign the Contributor License Agreement.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. I have one initial question: which implementation of CRYSTALS are you planning to use? Could you link to the GitHub repo? Also, what are your long-term goals with this project?

**Comment by andredif** [comment]
Hi David, thank you for your questions.
Regarding the Rust implementation in the Substrate node, I'm gonna use the
pq-crystals implementation: https://github.com/pq-crystals/dilithium
(Apache 2.0 license) ; while in the
browser extension, which requires Javascript, I was planning to use the npm
package https://www.npmjs.com/package/dilithium-crystals (MIT License) ,
adapting it to an HD wallet .
The long-term goal is to create a start-up for the project, and build a
full quantum-safe parachain. The idea is to use this Grant to get the first
bricks of the project, and then once
it's been shown the interest of the community around it, create my own
company and a small team to bring it to a testnet-level first and then to a
production-level blockchain.

Best regards,

Andrea Di Franco

Il giorno lun 20 nov 2023 alle ore 08:23 David Hawig <
***@***.***> ha scritto:

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the quick reply. Could you integrate this into the application? Also, feel free to add more information regarding your cryptography background or similar experience that you have.

**Comment by andredif** [comment]
Thank you, David. I have updated my applications with the repositories' links and a brief description of my experience with cryptography

**Review by Noc2 — DISMISSED** [review]
Thanks for the update. I'm happy to go ahead with it and mark it as ready for review.

**Review by takahser — CHANGES_REQUESTED** [review]
@andredif could you add the mandatory Docker deliverable to milestone 2 as well? Other than that I'd be willing to support this project as well.

File: applications/quantum-guard.md
Comment (takahser) [inline applications/quantum-guard.md:21]:
```suggestion
  - CRYSTALS-KYBER algorithm for key exchange;
  - CRYSTALS-Dilithium for asymmetric cryptography.
```

File: applications/quantum-guard.md
Comment (takahser) [inline applications/quantum-guard.md:17]:
```suggestion
- Use a quantum-resistant communication protocol between nodes.

On August 28th, 2023, the National Institute of Standards and Technology (NIST) issued an official document stating that 4 algorithms will be standardized as official PQC.
```

**Comment by andredif** [comment]
@takahser Thank you for your review and the corrections you've made. I've added the 0d-Dockerfile deliverable to Milestone2.

**Review by takahser — DISMISSED** [review]
@andredif thanks for the update, LGTM.

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Thanks @andredif our [payment options](https://github.com/w3f/Grants-Program#pencil-process) have also recently changed. Would you be willing to use USDC on Asset Hub instead of ETH mainnet?

**Comment by andredif** [comment]
Hi @keeganquigley , I apologize for not complying with new payment rules. I've just updated the application with my Polkadot Asset Hub address.

**Review by takahser — APPROVED** [review]

**Comment by semuelle** [comment]
Hi @andredif, sorry for the inconvenience, but we [just started requiring KYC/KYB checks](https://grants.web3.foundation/docs/faq#what-is-kyckyb-and-why-do-i-have-to-provide-this-data) for all potential grantees. Could you please provide the information outlined under [this link](https://in.sumsub.com/idensic/l/#/uni_mjcB05gJZfh8dbr7)? Let me know if you have any questions or issues.

**Comment by andredif** [comment]
Hi @semuelle , I can confirm you that I've successfully completed the KYC procedure

**Review by Noc2 — APPROVED** [review]

**Comment by andredif** [comment]
Thank you all for approving my Grant application. I will read carefully the Milestone delivery instructions and start working on the project in the next upcoming days.

**Comment by andredif** [comment]
Dear all,
due to the complexity of the task regarding the first milestone, and other issues, I'd kindly ask you to extend the delivery date of the first milestone up to 3 more working weeks, roughly to the beginning of March. Would it be possible?

**Comment by keeganquigley** [comment]
Hi @andredif thanks for your inquiry. Sure, that shouldn't be a problem since the project technically isn't due for a couple more weeks. If you come to find that it will be delayed further than 4 weeks, we would simply ask you to submit an amendment to the original application to extend the timeline. Thanks for your vigilance in this matter!

**Comment by andredif** [comment]
Thank you for the explanation, I really appreciate it! I'll be working hard to complete the milestone in the shortest time possible.

**Comment by keeganquigley** [comment]
Hi @andredif just checking in to see if you are on track to deliver soon. If not feel free to submit an amendment.

**Comment by andredif** [comment]
Hi @keeganquigley yes, sadly I faced many caveats while working on the project, and I definitely need to amend the delivery date. What is the correct procedure? Do I need to rewrite the total weeks needed for the milestones, recalculating them from the start of the project till the new expected date of delivery?

**Comment by keeganquigley** [comment]
Hi @andredif thanks for the update. Yes, you can submit a pull request to amend the original [application](https://github.com/w3f/Grants-Program/blob/master/applications/quantum-guard.md) to extend the timeline, [see here](https://github.com/w3f/Grants-Program?tab=readme-ov-file#changes-to-a-grant-after-approval) for full details. This will need 2 approvals from the committee but is usually passed fairly quickly. Please update the total weeks needed as well as each milestone for estimated delivery date. Thanks!

**Comment by andredif** [comment]
hi @keeganquigley , I've created a pull request with the amended time schedule. Thank you again for the opportunity

**Comment by keeganquigley** [comment]
Hi @andredif we understand that you have other obligations, but at the same time, we want to make sure the grant is still being worked on. Please let us know otherwise we will close it for now and you can potentially re-open it in the future. Thanks.

**Comment by andredif** [comment]
Hi Keegan,
I deeply apologies for not having delivered the project in time. Many
problems have arose. If you'd give me the opportunity to block the grant
and try to re-open it in the future, I believe this could be the best
option at the moment.

Best regards,

Andrea Di Franco

Il giorno gio 7 nov 2024 alle ore 22:46 Keegan | W3F <
***@***.***> ha scritto:

**Comment by keeganquigley** [comment]
Sounds good @andredif thanks for the update and sorry to hear about the difficulties. Let's do that for now, and you can let us know if you'd like to re-open it in the future. It would just require two approvals from the committee again. Thanks!

**Comment by andredif** [comment]
Dear all,
I'd love to explore the possibility to resume the work on the QuantumGuard project and deliver the milestone in the next future. Would it still be possible to resume the application ?
Best regards,

Andrea

**Pull request merged**
